syntax = "proto3";

package GameController.Leagues.SPL.Proto;

enum CompetitionPhase {
  COMPETITION_PHASE_ROUNDROBIN = 0;
  COMPETITION_PHASE_PLAYOFF = 1;
};

enum CompetitionType {
  COMPETITION_TYPE_NORMAL = 0;
}

enum GamePhase {
  GAME_PHASE_FIRST_HALF = 0;
  GAME_PHASE_SECOND_HALF = 1;
  GAME_PHASE_PENALTY_SHOOTOUT = 2;
};

enum GameState {
  GAME_STATE_INITIAL = 0;
  GAME_STATE_READY = 1;
  GAME_STATE_SET = 2;
  GAME_STATE_PLAYING = 3;
  GAME_STATE_FINISHED = 4;
};

enum SetPlay {
  SET_PLAY_NONE = 0;
  SET_PLAY_GOAL_KICK = 1;
  SET_PLAY_PUSHING_FREE_KICK = 2;
  SET_PLAY_CORNER_KICK = 3;
  SET_PLAY_KICK_IN = 4;
  SET_PLAY_PENALTY_KICK = 5;
};

enum TeamColor {
  TEAM_COLOR_BLUE = 0;
  TEAM_COLOR_RED = 1;
  TEAM_COLOR_YELLOW = 2;
  TEAM_COLOR_BLACK = 3;
  TEAM_COLOR_WHITE = 4;
  TEAM_COLOR_GREEN = 5;
  TEAM_COLOR_ORANGE = 6;
  TEAM_COLOR_PURPLE = 7;
  TEAM_COLOR_BROWN = 8;
  TEAM_COLOR_GRAY = 9;
};

enum Penalty {
  PENALTY_NONE = 0;
  PENALTY_REQUEST_FOR_PICKUP = 1;
  PENALTY_ILLEGAL_POSITION = 2;
  PENALTY_ILLEGAL_POSITION_IN_SET = 3;
  PENALTY_MOTION_IN_SET = 4;
  PENALTY_FALLEN_INACTIVE = 5;
  PENALTY_LOCAL_GAME_STUCK = 6;
  PENALTY_BALL_HOLDING = 7;
  PENALTY_PLAYER_STANCE = 8;
  PENALTY_PLAYER_PUSHING = 9;
  PENALTY_PLAYING_WITH_HANDS = 10;
  PENALTY_LEAVING_THE_FIELD = 11;
  PENALTY_SUBSTITUTE = 12;
};
